luxonis_train/nodes/backbones/ddrnet/ddrnet.py

```diff
@@ -0,0 +1,295 @@
+            expansion=layer5_bottleneck_expansion,
+        )
+
+        self.layer5 = _make_layer(
+            block=layer5_block,
+            in_planes=out_chan_backbone["layer4"],
+            planes=out_chan_backbone["layer4"],
+            num_blocks=self.additional_layers[0],
+            stride=2,
+            expansion=layer5_bottleneck_expansion,
+        )
+
+        self.spp = DAPPM(
+            in_planes=out_chan_backbone["layer4"] * layer5_bottleneck_expansion,
+            branch_planes=spp_width,
+            out_planes=highres_planes * layer5_bottleneck_expansion,
+            inter_mode=self.ssp_inter_mode,
+            kernel_sizes=spp_kernel_sizes,
+            strides=spp_strides,
+        )
+
+        self.highres_planes = highres_planes
+        self.layer5_bottleneck_expansion = layer5_bottleneck_expansion
+        self.init_params()
+
+    @property
+    def backbone(self):
+        """Create a fake backbone module to load backbone pre-trained weights."""
+        return nn.Sequential(
+            Dict(
+                [
+                    ("_backbone", self._backbone),
+                    ("compression3", self.compression3),
+                    ("compression4", self.compression4),
+                    ("down3", self.down3),
+                    ("down4", self.down4),
+                    ("layer3_skip", self.layer3_skip),
+                    ("layer4_skip", self.layer4_skip),
+                    ("layer5_skip", self.layer5_skip),
+                ]
+            )
+        )
+
+    def forward(self, inputs: Tensor) -> list[Tensor]:
+        width_output = inputs.shape[-1] // 8
+        height_output = inputs.shape[-2] // 8
+
+        x = self._backbone.stem(inputs)
+        x = self._backbone.layer1(x)
+        x = self._backbone.layer2(self.relu(x))
+
+        # Repeat layer 3
+        x_skip = x
+        for i in range(self.layer3_repeats):
+            out_layer3 = self._backbone.layer3[i](self.relu(x))
+            out_layer3_skip = self.layer3_skip[i](self.relu(x_skip))
+
+            x = out_layer3 + self.down3[i](self.relu(out_layer3_skip))
+            x_skip = out_layer3_skip + self.upscale(
+                self.compression3[i](self.relu(out_layer3)), height_output, width_output
+            )
+
+        # Save for auxiliary head
+        if self._use_aux_heads:
+            x_extra = x_skip
+
+        out_layer4 = self._backbone.layer4(self.relu(x))
+        out_layer4_skip = self.layer4_skip(self.relu(x_skip))
+
+        x = out_layer4 + self.down4(self.relu(out_layer4_skip))
+        x_skip = out_layer4_skip + self.upscale(
+            self.compression4(self.relu(out_layer4)), height_output, width_output
+        )
+
+        out_layer5_skip = self.layer5_skip(self.relu(x_skip))
+
+        x = self.upscale(
+            self.spp(self.layer5(self.relu(x))), height_output, width_output
+        )
+
+        x = x + out_layer5_skip
+
+        if self._use_aux_heads:
+            return [x, x_extra]
```

Comment:
The order would be better reversed (`[x_extra, x]`) for a better compatibility with different heads. See the comment under `attach_index` for `DDRNetSegmentationHead`